src/pudl/transform/eia.py

```diff
@@ -37,14 +37,15 @@
 
 import pudl
 from pudl.helpers import convert_cols_dtypes
-from pudl.metadata.classes import Package
 from pudl.metadata.enums import APPROXIMATE_TIMEZONES
 from pudl.metadata.fields import apply_pudl_dtypes, get_pudl_dtypes
 from pudl.metadata.resources import ENTITIES
 from pudl.settings import EiaSettings
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
+PUDL_PKG = pudl.metadata.classes.Package.from_resource_ids()
```

Comment:
Compiling the dictionary of encoders only takes 50ms (1/20th of a second) but it does seem silly to do it over and over again, and in general the pattern we have now of recreating the Package from scratch everywhere many times seems kind of silly. This doesn't have to be defined here, but as an example it works.

Reply:
I don't think this package actually compiles the dictionary. If i understand correctly the `cached_property` only compiles the dictionary the first time the property is accessed

Reply:
if there is no variation in all of the gazillion places where we are currently recreating the Package.... should we just define this in `metadata.classes` or somewhere in metadata and use that across the board?

Reply:
There's like 50 places where we create the full Package and they're always exactly the same and imutable. I'd love to put this in the `metadata.classes` and just reference it there from everywhere. It'll have to be at the bottom after the `Package` has been defined but that's fine.